api/v2/clairpb/clair.proto

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 clair authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+option go_package = "clairpb";
+
+package clairpb;
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+
+message Vulnerability{
+	string name = 1;
+	string namespace_name = 2;
+	string description = 3;
+	string link = 4;
+	string severity = 5;
+	string metadata = 6;
+	string fixed_by = 7;
+	repeated Feature fixed_in = 8;
+}
+
+message Feature{
+	string name = 1;
+	string namespace_name = 2;
+	string version = 3;
+	string version_format = 4;
+	string added_by = 5;
+	repeated Vulnerability vulnerabilities = 6;
+}
+
+message AncestryLayer{
+	string name = 1;
+	string path = 2;
+	map<string, string> headers = 3;
+}
+
+message Ancestry{
+	string name = 1;
+	string format = 2;
+	repeated AncestryLayer layers = 3;
+}
+
+message LayersIntroducingVulnerabilty{
+	Vulnerability vulnerability = 1;
+	repeated OrderedLayerName ordered_layers_introducing_vulnerability = 2;
+}
+
+message OrderedLayerName{
+	int32 index = 1;
+	string layer_name = 2;
+}
+
+message Layer{
+	string name = 1;
+	repeated string namespace_names = 2;
+	string parent_name = 3;
+	repeated Feature features = 4;
+}
+
+message Notification{
+	string name = 1;
+	string created = 2;
+	string notified = 3;
+	string deleted = 4;
+	int32 limit = 5;
+	Page page = 6;
+}
+
+message Page{
+	string this_token = 1;
+	string next_token = 2;
+	LayersIntroducingVulnerabilty old = 3;
+	LayersIntroducingVulnerabilty new = 4;
+}
+
+message PostAncestryRequest{
+	Ancestry ancestry = 2;
+}
+
+message GetAncestryRequest{
+	string ancestry_name = 1;
+	int32  index = 2;
+	bool with_vulnerabilities = 3;
+	bool with_features = 4;
+}
+
+message GetNotificationRequest{
+	string page = 1;
+	int32 limit = 2;
+	string name = 3;
+}
+
+message DeleteNotificationRequest{
+	string name = 1;
+}
+
+message PostAncestryResponse{
+	Ancestry ancestry = 1;
+	int32 engine_version = 2;
+}
+
+message GetAncestryResponse{
+	Layer layer = 1;
+	int32 engine_version = 2;
+}
+
+message GetNotificationResponse{
+	Notification notification = 1;
+}
+
+service ClairAPIService{
```

Comment:
I talked to the etcd team and I think it's a nice pattern to separate Services by resource, so we can have two Services that we register to the same grpc server.

```protobuf
service AncestryService {
   rpc GetAncestry(GetAncestryRequest) returns (GetAncestryResponse) { ... };
   rpc PostAncestry(PostAncestryRequest) returns (PostAncestryResponse) { ... };
}

service NotificationService {
  rpc GetNotification(GetNotificationRequest) returns (GetNotificationResponse) { ... };
  rpc DeleteNotification(DeleteNotificationRequest) returns (DeleteNotificationResponse) { ... };
}
```